### PR TITLE
Fixing lib coredump

### PR DESCRIPTION
**Pull request by docbrown1955**

### CODE REVIEW

File: src/csl/libraryevaluator.cpp [hunk 1]
```diff
@@ -519,8 +519,9 @@ LibEval exprToEval(Expr const &expr, std::vector<LibEval> const &eval)
 {
     std::string_view name = expr->getName();
     char             buffer[5]{0, 0, 0, 0, 0};
-    for (size_t i = LibEval::name.size() + 1; i != name.size(); ++i)
-        buffer[i - LibEval::name.size() - 1] = name[i];
+    for (size_t i = LibEval::name.size() + 1; i != name.size() && i - LibEval::name.size() - 1 < 4; ++i)
+        if (i < name.size())
+            buffer[i - LibEval::name.size() - 1] = name[i];
```
Comment (docbrown1955) [inline src/csl/libraryevaluator.cpp:524]: Is the issue here that `name.size()` can be smaller than the first loop index value `LibEval::name.size() + 1;` such that the condition `i != name.size()`?

Not sure really how that could happen or why it would not have been an issue before.

Anyway, it's better to just change the condition in the for loop to `i < name.size()` instead of adding one more in the body

File: src/csl/libraryevaluator.cpp [hunk 2]
```diff
@@ -593,16 +594,17 @@ void LibEvalSession::getPerf(LibEval const        &init,
...
+        Expr recursive = init.init;
+        csl::VisitEachLeaf(recursive, [&](Expr const &expr) {
+            if (LibEval::isInstance(expr))
+                getPerf(exprToEval(expr, eval), newEvals, nOccurences);
+        });
     }
     else {
         ++nOccurences[std::distance(newEvals.begin(), pos)];
+        // Already visited, don't recurse again to avoid infinite recursion
     }
```
Comment (docbrown1955) [inline src/csl/libraryevaluator.cpp:607]: This is ok.

Not sure why that worked before, probably because an abbreviation should never be a leaf (because in that case there is nothing to be abbreviated), and maybe some strange edge case was hidding somewhere

Anyway, the change looks good so if the tests pass I'm ok

**Review by docbrown1955 — COMMENTED** [review]
@marco-palmiotto @josgonalves

I created that PR from your branch solving #87 

See my comments, minor change requested - otherwise it looks good (although the original issue is not clear)

**Comment by marco-palmiotto** [comment]
Hi, we had an issue with that function as well, so I'd say that this pull request can safely be ignored. We found out that the bug was probably coming from the source code not being compiled with -std=c++20, but std=c++17. At this point, I'd propose just to update the documentation, and probably the bug will be gone.
